docs/non-normative/how-to-write-conventions/resource-and-entities.md

```diff
@@ -0,0 +1,331 @@
+# Resource and Entities
+
+<!-- toc -->
+
+- [Modelling Guide](#modelling-guide)
+  - [Declaring associations between signals](#declaring-associations-between-signals)
+  - [Extending an entity](#extending-an-entity)
+- [FAQ](#faq)
+  - [When to define a new entity?](#when-to-define-a-new-entity)
+  - [What is an "is-a" relationship?](#what-is-an-is-a-relationship)
+  - [When to define an "is-a" relationship vs. extending descriptive attributes?](#when-to-define-an-is-a-relationship-vs-extending-descriptive-attributes)
+  - [How to define identifying attributes?](#how-to-define-identifying-attributes)
+    - [Multi-Observer Guidance](#multi-observer-guidance)
+  - [How to namespace entities?](#how-to-namespace-entities)
+- [Background: Resource and Entities](#background-resource-and-entities)
+  - [Open Expansion](#open-expansion)
+  - [Telescoping Identity](#telescoping-identity)
+
+<!-- tocstop -->
+
+## Modelling Guide
+
+To define an entity, create a new semantic convention model file, with a group type as `entity`, for example:
+
+`model/{my_domain}/entities.yaml`:
+
+```yaml
+groups:
+  - id: entity.my_entity
+    type: entity
+    stability: development
+    name: my_entity
+    brief: >
+      A description of my_entity here.
+    attributes:
+      - ref: some.attribute
+        role: identifying
+      - ref: some.other_attribute
+        role: descriptive
+        ...
+```
+
+Here, the attributes field contains all attributes of the Entity.
+The `role` of each attribute determines if it is identifying or
+descriptive. See
+[How to define identifying attributes?](#how-to-define-identifying-attributes)
+for details on what these mean.
+
+> Note: Declaring Entity relationships is not yet supported.
+
+### Declaring associations between signals
+
+You can declare which entities should be used with specific observability
+signals.  For example, process metrics should be used with the process
+entity, so that the metric is associated with a known process. To declare
+this, use the `entity_associations` field on the signal and reference
+another resource group *by name*.
+
+`model/{my_domain}/metrics.yaml`:
+
+```yaml
+groups:
+  - id: metric.some_metric
+    type: metric
+    ...
+    entity_associations: 
+      - my_entity
+```
+
+Notes:
+
+- You cannot declare an association on an *unstable* resource from a
+  *stable* signal.
+- You can declare multiple associations. These form a "one or many" set,
+  where one or many of the named entities may be associated with the
+  metric. There is *no* requirement to have one and only one entity
+  attached to a signal.
+
+### Extending an entity
+
+While not recommended for Semantic Conventions, you can define a new
+"view" of an entity that includes additional descriptive attributes. To
+do so, use the extends field on groups:
+
+`model/{my_other_domain}/entities.yaml`:
+
+```yaml
+groups:
+  - id: entity.my_entity_2
+    type: entity
+    extends: entity.my_entity
```

Comment:
This should probably be a list given the examples of views that we have "k8s workload" or "cloud"